options.go

```diff
@@ -0,0 +1,141 @@
+package configparser
+
+import (
+	"strings"
+
+	"github.com/bigkevmcd/go-configparser/chainmap"
+)
+
+const defaultSectionName = "DEFAULT"
+
+// options allows to control parser behavior.
+type options struct {
+	interpolation         Interpolator
+	commentPrefixes       Prefixes
+	inlineCommentPrefixes Prefixes
+	defaultSection        string
+	delimiters            string
+	converters            Converter
+	allowNoValue          bool
+	strict                bool
+
+	// TODO: under investigation, have no effect now.
+	emptyLines bool
+}
+
+// Converter contains custom convert functions for available types.
+//
+// List of types:
+//   - int -> int64
+//   - float -> float64
+//   - string
+//   - bool
+//
+// The caller should guarantee type assertion to the requested type
+// after custom processing!
+type Converter map[string]ConvertFunc
+
+// ConvertFunc is a custom datatype converter.
+type ConvertFunc func(string) (any, error)
+
+// Prefixes stores available prefixes for comments.
+type Prefixes []string
+
+// AtFirst checks if str starts with any of the prefixes.
+func (pr Prefixes) AtFirst(str string) bool {
+	for _, p := range pr {
+		if strings.HasPrefix(str, p) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Split splits str with the first prefix found.
+// Returns original string if no matches.
+func (pr Prefixes) Split(str string) string {
+	for _, p := range pr {
+		if strings.Contains(str, p) {
+			return strings.Split(str, p)[0]
```

Comment:
Given that these are prefixes, should this use `HasPrefix` ?

Reply:
This function is used for inline comments e.g. `option = value # this is a comment`. Thus, here it checks if given string contains any of the `Prefixes`